addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV6SessionManager.java

```diff
@@ -331,8 +351,10 @@ public void keep_alive(int periodic_interval_ms) {
     /**
      * The main state machine of the session handshake. This will start a timer
      * that will reset the handshake if we do not get a satisfying response in time.
+     *
+     * @throws InterruptedException
      */
-    private void session_handshake_process() {
+    private void session_handshake_process() throws InterruptedException {
```

Comment:
No that is not what I meant I was too lazy to check whether each added ` throws InterruptedException` was really needed :-D 

I agree on the concept to expose it to the caller so you can properly stop what you are doing.